preprocessing/science/refine_srf.py

```diff
@@ -0,0 +1,210 @@
+                print("srf version: %s not supported" % (line))
+                raise
+            # skip comments
+            while True:
+                line = fid.readline()
+                if not line.startswith("#"):
+                    break
+            line_el = line.split()
+            if line_el[0] != "PLANE":
+                print("no plane specified")
+                raise
+            if line_el[1] != "1":
+                print("only one plane supported")
+                raise
+            line_el = fid.readline().split()
+            nx = int(line_el[2])
+            ny = int(line_el[3])
+            line_el = fid.readline().split()
+            # check that the plane data are consistent with the number of points
+            assert int(line_el[1]) == nx * ny
+            self.init_spatial_arrays(nx, ny)
+            for j in range(ny):
+                for i in range(nx):
+                    # first header line
+                    line = fid.readline()
+                    if abs(version - 1.0) < 1e-03:
+                        self.lon[j, i], self.lat[j, i], self.depth[j, i], self.strike[j, i], self.dip[j, i], self.PSarea, self.t0[j, i], dt = [float(v) for v in line.split()]
+                    else:
+                        # srf version 2 has also density and Vs in that line
+                        self.lon[j, i], self.lat[j, i], self.depth[j, i], self.strike[j, i], self.dip[j, i], self.PSarea, self.t0[j, i], dt = [float(v) for v in line.split()[0:-2]]
+                    # second header line
+                    line = fid.readline()
+                    self.rake[j, i], self.slip1[j, i], ndt1, slip2, ndt2, slip3, ndt3 = [float(v) for v in line.split()]
+                    if max(slip2, slip3) > 0.0:
+                        print("this script assumes slip2 and slip3 are zero", slip2, slip3)
+                        raise
+                    ndt1 = int(ndt1)
+                    if max(i, j) == 0:
+                        self.ndt = ndt1
+                        self.dt = dt
+                        self.init_aSR()
+                    lSTF = []
+                    if ndt1 == 0:
+                        continue
+                    if ndt1 > self.ndt:
+                        print("a larger ndt (%d> %d)was found for point source (i,j) = (%d, %d) extending aSR array..." % (ndt1, self.ndt, i, j))
+                        self.extend_aSR(self.ndt, ndt1)
+                    if abs(dt - self.dt) > 1e-6:
+                        print("this script assumes that dt is the same for all sources", dt, self.dt)
+                        raise
+                    while True:
+                        line = fid.readline()
+                        lSTF.extend(line.split())
+                        if len(lSTF) == ndt1:
+                            self.aSR[j, i, 0:ndt1] = np.array([float(v) for v in lSTF])
+                            break
+
+    def upsample_fault(self, p, spatial_order, spatial_zoom, temporal_zoom):
```

Comment:
I'd reverser the order: p2 = p.upsample_fault(...)

Reply:
adressed by last commits?